library/asb_set_state.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/python
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.0',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+
+
+import os
+
+from ansible.module_utils.basic import AnsibleModule
+try:
+    from kubernetes import client, config
+    from kubernetes.client.rest import ApiException
+    config.load_kube_config()
+    api = client.CoreV1Api()
+except Exception as error:
+    ansible_module.fail_json(
+        msg="Error attempting to load kubernetes client: {}".format(error))
+
+
+ENV_NAME = "POD_NAME"
+ENV_NAMESPACE = "POD_NAMESPACE"
+
+
+def create_config_map(name, namespace, data):
+    meta = client.V1ObjectMeta(name=name)
+    body = client.V1ConfigMap(metadata=meta, data=data)
+    try:
+        res = api.create_namespaced_config_map(namespace, body)
+    except ApiException as e:
+        module.fail_json(
+            msg="Error attempting to create namespaced configmap: {}".format(e))
+
+
+def update_config_map(name, namespace, data):
+    body = client.V1ConfigMap(data=data)
+    try:
+        res = api.patch_namespaced_config_map(name, namespace, body)
+    except ApiException as e:
+        module.fail_json(
+            msg="Error attempting to update configmap: {}".format(e))
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            key=dict(required=True, type='str'),
+            value=dict(required=True, type='str'),
+        ),
+    )
+
+    try:
+        name = os.environ[ENV_NAME]
+        namespace = os.environ[ENV_NAMESPACE]
+    except Exception as e:
+        module.fail_json(
+            msg="Error attempting to get name/namespace from environment: {}".format(e))
+
+    data = {module.params['key']: module.params['value']}
+
+    try:
+        config_map = api.read_namespaced_config_map(name, namespace)
+    except ApiException as e:
+        if e.status == 404:
+            create_config_map(name, namespace, data)
+        else:
+            module.fail_json(
+                msg="Error attempting to read namespaced configmap: {}".format(e))
+
+    update_config_map(name, namespace, data)
+    module.exit_json(changed=True)
```

Comment:
This always returns changed status, other option I think is to read the existing config map and see if we are overwriting anything to return correct status but not sure if this is required?